### PR TITLE
Create incus user and manage userns subuid/gid

### LINKED ISSUE #14: Consider creating `incus` system user as part of package
Hello,

During some tests of these packages, I could see that there is some groups created for Incus, but there is no `incus` system user created for sub-processes such as dnsmasq or QEMU. Therefore, these processes uses `nobody` user for their exec, I was a bit surprised at first coming from LXD where there is a native `lxd` user. After looking in the Incus source code, `nobody` user is indeed used if there is no `incus` user present on the system. In fact, I think this doesn't change anything, I just think this is a bit cleaner from a system administration perspective.

WDYT ? I can prepare a PR if needed :)

### LINKED ISSUE #15: Management of subuid/subgid as part of package
Hello !

I was looking at the [LXD Debian package sources](https://salsa.debian.org/go-team/packages/lxd/-/blob/debian/sid/debian/lxd.preinst?ref_type=heads) and see that when user are created, the `lxd` and `root` users get some subuid/gid for user namespacing stuff. 

Moreover in the docs, when building it from sources, it is mentioned that we need to add at least 10M range of UID/GID for the `root` user:
https://linuxcontainers.org/incus/docs/main/installing/#machine-setup

I don't see anything related to this in the Incus package, do we still need to handle it as part of package or there is some new kernel stuff that manage it automatically ?

Thanks !

### PR DESCRIPTION
**Pull request by f-bn**
Closes #14, closes #15

### CODE REVIEW

**Comment by stgraber** [comment]
Looks good, thanks!

**Comment by stgraber** [comment]
Hmm, actually, I wonder if there's a way to not require `uidmap`?

The behavior when it's not present tends to be better. It makes sense to have the allocations in place for when it is present, but I don't think we should force users to get it.

The main case I can think of is `incus-user` as that allows individual users to map their own uid/gid into the container. This won't work with subuid/subgid without adding another root allocation for the user's own uid and gid.

**Comment by f-bn** [comment]
I'm not the most confident with these configurations, so I can add a simple condition to manage these subuid/subgid based on `uidmap` package presence. Something like :

```
if ! dpkg -l | grep -q uidmap; then
    # manage subuid/gid
fi
```

I can also remove this part if not strictly needed and can cause issues. In this case, we may need to clarify the documentation about it, I don't have a strong opinion on this one.

WDYT ?

**Comment by stgraber** [comment]
usermod should still work fine even if uidmap isn't installed so I'd just not make it a dependency

File: debian/incus.preinst [hunk 1]
```diff
@@ -10,6 +10,38 @@ case "$1" in
         if ! getent group incus-admin >/dev/null; then
             addgroup --system incus-admin
         fi
+
+        if ! getent passwd incus >/dev/null; then
+            adduser --system --shell /usr/sbin/nologin --home /var/lib/incus --ingroup incus incus
```
Comment (stgraber) [inline debian/incus.preinst]: We'll want `--no-create-home` too I think.

**Comment by stgraber** [comment]
I've confirmed that the subuid/subgid usermod logic works perfectly fine when the `uidmap` package isn't installed. So the logic can stay as it is and just remove the `uidmap` dependency from debian/control will give us what we want.

File: debian/incus.preinst (hunk 1 above)
Comment (f-bn) [reply debian/incus.preinst]: Alright, will add it

**Comment by f-bn** [comment]
@stgraber requested changes pushed 👍

**Review by stgraber — APPROVED** [review]
Perfect! I gave it a quick rebase to merge the fixes with the original commits and it's now ready to merge.

Thanks!

**Pull request merged**